### PR TITLE
GNT Driver new API

### PR DESCRIPTION
**Pull request by bidzyyys**

### CODE REVIEW

**Comment by bidzyyys** [comment]
@Wiezzel @maaktweluit please do CR but again I have a problem with running demo scenario...

File: core/payment-driver/gnt/Cargo.toml [hunk 1]
```diff
@@ -18,10 +12,10 @@ awc = "1.0.1"
 bigdecimal = "0.1.0"
 bitflags = "1.2"
 chrono = { version = "0.4", features = ["serde"] }
-diesel = { version = "1.4", features = ["sqlite", "r2d2", "chrono"] }
-diesel_migrations = "1.4"
 ethereum-tx-sign = { git = "https://github.com/bidzyyys/ethereum-tx-sign" }
 ethereum-types = "0.6.0"
+diesel = { version = "1.4", features = ["sqlite", "r2d2", "chrono"] }
+diesel_migrations = "1.4"
```
Comment (Wiezzel) [inline core/payment-driver/gnt/Cargo.toml:18]: Why did you move these? They were alphabetically sorted.

File: core/payment-driver/gnt/src/dao/payment.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+impl<'c> AsDao<'c> for PaymentDao<'c> {
+    fn as_dao(pool: &'c PoolType) -> Self {
+        Self { pool }
+    }
+}
+
+impl<'c> PaymentDao<'c> {
+    pub async fn get_pending_payments(&self, address: String) -> DbResult<Vec<PaymentEntity>> {
+        do_with_transaction(self.pool, move |conn| {
```
Comment (Wiezzel) [inline core/payment-driver/gnt/src/dao/payment.rs:24]: When only reading better use `readonly_transaction`.

File: core/payment-driver/gnt/src/dao/payment.rs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            diesel::update(dsl::gnt_driver_payment.find(order_id))
+                .set((dsl::tx_id.eq(tx_id), dsl::status.eq(PAYMENT_STATUS_OK)))
+                .execute(conn)?;
+            Ok(())
+        })
+        .await
+    }
+
+    pub async fn get_by_tx_id(&self, tx_id: String) -> DbResult<Vec<PaymentEntity>> {
+        do_with_transaction(self.pool, move |conn| {
```
Comment (Wiezzel) [inline core/payment-driver/gnt/src/dao/payment.rs:66]: `readonly_transaction`

File: core/payment-driver/gnt/src/gnt/sender.rs [hunk 4]
```diff
@@ -917,3 +919,40 @@ async fn transfer_gnt(
...
+    if tx.tx_type != TRANSFER_TX {
+        return Ok(());
+    }
+
+    let payments = db.as_dao::<PaymentDao>().get_by_tx_id(tx_id).await?;
+    assert_ne!(payments.len(), 0);
+
+    let mut amount = BigDecimal::zero();
+    for payment in payments.iter() {
+        amount = amount.add(
```
Comment (Wiezzel) [inline core/payment-driver/gnt/src/gnt/sender.rs:939]: `+=` should do

File: Cargo.toml [hunk 5]
```diff
@@ -27,8 +27,8 @@ ya-market-decentralized = { version = "0.1", optional = true }
 ya-market-forwarding = { version = "0.1", optional = true }
 ya-net = { version = "0.1", features = ["service"] }
 ya-payment = "0.1"
-ya-payment-driver = "0.1"
 ya-dummy-driver = { version = "0.1", optional = true }
+ya-gnt-driver = { version = "0.1" }
```
[the remark this answers is not in the document]
Comment (bidzyyys) [reply Cargo.toml:31]: @Wiezzel ?

[the remark this answers is not in the document]
Comment (Wiezzel) [reply Cargo.toml:31]: That would require creating a new feature for gnt driver. Now we're using dummy and not-dummy. I'm not sure if we should mess with this now.

File: core/payment-driver/gnt/Cargo.toml [hunk 6]
```diff
@@ -18,10 +12,10 @@ awc = "1.0.1"
 bigdecimal = "0.1.0"
 bitflags = "1.2"
 chrono = { version = "0.4", features = ["serde"] }
-diesel = { version = "1.4", features = ["sqlite", "r2d2", "chrono"] }
-diesel_migrations = "1.4"
 ethereum-tx-sign = { git = "https://github.com/bidzyyys/ethereum-tx-sign" }
 ethereum-types = "0.6.0"
+diesel = { version = "1.4", features = ["sqlite", "r2d2", "chrono"] }
```
[the remark this answers is not in the document]
Comment (bidzyyys) [reply core/payment-driver/gnt/Cargo.toml:17]: My fault ;)
